### DISCUSSION TITLE
How to get Bounds, Width, etc. before first Render(...) call?

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#9966] [General]
I need to calculate some stuff based on the control size, ideally before the first time my `public override void Render(...)` is called.  I tried checking in `OnInit(...)` and also with `AttachedToVisualTree(...)` but in both of these the `Bounds` property of my `TemplatedContral` is still `0,0,0,0`. 

I'd prefer NOT to set a bool on the first call to Render to get these values.  I think getting `Bounds` on some resize or relayout event would be useful and probably a proper place for this calculation. Is `OnLayoutUpdated` the correct handler to which to add my calc?

thanks

**Comment by amwx** [comment]
Both `OnInitialized` and `AttachedToVisualTree` are handled before any layout passes occur, so that's why Bounds is still empty. 

Yes `LayoutUpdated` should work. Alternatively, you could also override `ArrangeOverride` and let the base method run. The return result there should be the new size what is set to bounds. `ArrangeCore` I believe is similar. Both of these are the earliest the control will know its layout size and location. `LayoutUpdated` is raised when that layout pass completes.
